ml-agents/mlagents/trainers/ghost/controller.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import Deque, Dict
+from collections import deque
+from mlagents.trainers.ghost.trainer import GhostTrainer
+
+logger = logging.getLogger("mlagents.trainers")
+
+
+class GhostController(object):
+    """
+    GhostController contains a queue of team ids. GhostTrainers subscribe to the GhostController and query
+    it to get the current learning team.  The GhostController cycles through team ids every 'swap_interval'
+    which corresponds to the number of trainer steps between changing learning teams.
```

Comment:
Please mention this is a unique object and there can only be one in the training session.